ibridgesgui/popup_widgets.py

```diff
@@ -69,7 +76,114 @@ def accept(self):
                 os.makedirs(new_dir_path)
                 self.done(1)
             except Exception as error:
-                if hasattr(error, 'message'):
+                if hasattr(error, "message"):
                     self.errorLabel.setText(error.message)
                 else:
                     self.errorLabel.setText("ERROR: insufficient rights.")
+
+#class CheckConfig(QDialog, Ui_configCheck):
+class CheckConfig(QDialog):
+    """Popup window to edit, create and check an environment.json"""
+    def __init__(self, logger, env_path):
+        super().__init__()
+        if getattr(sys, "frozen", False):
+            super().setupUi(self)
+        else:
+            loadUi(UI_FILE_DIR / "configCheck.ui", self)
+
+        self.logger = logger
+        self.env_path = env_path
+        self.setWindowTitle("Create, edit and inspect iRODS environment")
+        self.setWindowFlags(QtCore.Qt.WindowType.WindowStaysOnTopHint)
+        self._init_env_box()
+
+        self.envbox.activated.connect(self.load_env)
+        self.createButton.clicked.connect(self.create_env)
+        self.checkButton.clicked.connect(self.check_env)
+        self.saveButton.clicked.connect(self.save_env)
+        self.saveasButton.clicked.connect(self.save_env_as)
+        self.closeButton.clicked.connect(self.close)
+
+
+    def _init_env_box(self):
+        self.envbox.clear()
+        env_jsons = [""]+[
+            path.name for path in
+            self.env_path.glob("irods_environment*json")]
+        if len(env_jsons) != 0:
+            self.envbox.addItems(env_jsons)
+            self.envbox.setCurrentIndex(0)
+
+    def load_env(self):
+        """Load json into text field."""
+        self.errorLabel.clear()
+        env_file = self.env_path.joinpath(self.envbox.currentText())
+        try:
+            content = json.dumps(_read_json(env_file),
+                                sort_keys=True, indent=4, separators=(",", ": "))
+            populate_textfield(self.envEdit, content)
+        except IsADirectoryError:
+            self.errorLabel.setText("Choose and environment or create a new one.")
+        except FileNotFoundError:
+            self.errorLabel.setText(f"File does not exist {env_file}")
+        except Exception as err:
+            self.errorLabel.setText(f"{repr(err)}")
+
+    def create_env(self):
+        """Load standard environment into text field"""
+        self.errorLabel.clear()
+        self.envbox.setCurrentIndex(0)
+        env = {
```

Comment:
We might want to make this less UU centric at some point.

Reply:
Right now this is okay of course.

Reply:
I will change it immediately ;)